docs/source/python/orc.rst

```diff
@@ -0,0 +1,261 @@
+Impala (incubating) <http://impala.apache.org>`_, and `Apache Spark
+<http://spark.apache.org>`_ adopting it as a shared standard for high
+performance data IO.
+
+Apache Arrow is an ideal in-memory transport layer for data that is being read
+or written with ORC files.
+
+Obtaining pyarrow with ORC Support
+--------------------------------------
+
+If you installed ``pyarrow`` with pip or conda, it should be built with ORC
+support bundled:
+
+.. ipython:: python
+
+   import pyarrow.orc as po
+
+If you are building ``pyarrow`` from source, you must use
+``-DARROW_ORC=ON`` when compiling the C++ libraries and enable the ORC
+extensions when building ``pyarrow``. See the :ref:`Python Development
+<python-development>` page for more details.
+
+Reading and Writing Single Files
+--------------------------------
+
+The functions :func:`~.orc.read_table` and :func:`~.orc.write_table`
+read and write the :ref:`pyarrow.Table <data.table>` object, respectively.
+
+Let's look at a simple table:
+
+.. ipython:: python
+
+   import numpy as np
+   import pandas as pd
+   import pyarrow as pa
+
+   df = pd.DataFrame({'one': [-1, np.nan, 2.5],
+                      'two': ['foo', 'bar', 'baz'],
+                      'three': [True, False, True]},
+                      index=list('abc'))
+   table = pa.Table.from_pandas(df)
+
+We write this to ORC format with ``write_table``:
+
+.. ipython:: python
+
+   import pyarrow.orc as po
+   po.write_table(table, 'example.orc')
+
+This creates a single ORC file. In practice, an ORC dataset may consist
+of many files in many directories. We can read a single file back with
+``read_table``:
+
+.. ipython:: python
+
+   table2 = po.read_table('example.orc')
+   table2.to_pandas()
+
+You can pass a subset of columns to read, which can be much faster than reading
+the whole file (due to the columnar layout):
+
+.. ipython:: python
+
+   po.read_table('example.orc', columns=['one', 'three'])
+
+We need not use a string to specify the origin of the file. It can be any of:
+
+* A file path as a string
+* A :ref:`NativeFile <io.native_file>` from PyArrow
+* A Python file object
+
+In general, a Python file object will have the worst read performance, while a
+string file path or an instance of :class:`~.NativeFile` (especially memory
+maps) will perform the best.
+
+ORC file writing options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`~pyarrow.orc.write_table()` has a number of options to
+control various settings when writing an ORC file.
+
+* ``file_version``, the ORC format version to use.  ``'0.11'`` ensures
+  compatibility with older readers, while ``'0.12'`` is the newer one.
+* ``stripe_size``, to control the approximate size of data within a column 
+  stripe. This currently defaults to 64MB.
+
+See the :func:`~pyarrow.orc.write_table()` docstring for more details.
+
+There are some additional data type handling-specific options
+described below.
+
+Omitting the DataFrame index
```

Comment:
I get the impression you're copying and adapting large chunks of the Parquet docs. I'm not sure it makes sense to do this (also, I don't know why the Parquet docs talk about `preserve_index` specifically).

Reply:
Yes, I think in this case we can omit this section about the pandas index (or maybe refer to the pandas.rst page on those details). 

(I suppose the reason that the parquet page includes those details is that, historically, many people where using the parquet methods to get pandas DataFrames (the ParquetFile also has functionality to directly get a DataFrame))